src/main/scala/org/dia/algorithms/mcc/NetcdfDFSMCC.scala

```diff
@@ -152,157 +132,80 @@ object NetcdfDFSMCC {
             /** Find non-zero points in product array */
+            updateComponent(components1(row, col), t1.metaData("FRAME"), t1.tensor(row, col), row, col)
+            updateComponent(components2(row, col), t2.metaData("FRAME"), t2.tensor(row, col), row, col)
             if (product(row, col) != 0.0) {
-              /** save components ids */
-              val value1 = components1(row, col)
-              val value2 = components2(row, col)
-              overlappedPairsList += ((value1, value2))
+
+              /** If overlap exists create and edge and update overlapped area */
+              val frame1 = t1.metaData("FRAME")
+              val label1 = components1(row, col)
+              val node1 = nodeMap(frame1 + ":" + label1)
+
+              val frame2 = t2.metaData("FRAME")
+              val label2 = components2(row, col)
+              val node2 = nodeMap(frame2 + ":" + label2)
+
+              val edgeKey = s"$frame1:$label1,$frame2:$label2"
+              val edge = if (MCCEdgeMap.contains(edgeKey)) MCCEdgeMap(edgeKey) else new MCCEdge(node1, node2)
+              edge.incrementAreaOverlap()
+              MCCEdgeMap.update(edgeKey, edge)
             }
-            updateComponent(components1(row, col), t1.metaData("FRAME"), t1.tensor(row, col))
-            updateComponent(components2(row, col), t2.metaData("FRAME"), t2.tensor(row, col))
           }
         }
 
-        /**
-         * This code essentially computes the number of times the same tuple occurred in the list,
-         * a repeat occurance would indicate that the components overlapped for more than one cell
-         * in the product matrix. By calculating the number of overlaps we can calculate the number of cells
-         * they overlapped for and since each cell is of a fixed area we can compute the area overlap
-         * between those two components.
-         */
-        var overlappedMap = overlappedPairsList.groupBy(identity).mapValues(_.size)
-        println(s"Overlap Map ${overlappedMap.size}")
+        val filtered = MCCEdgeMap.filter({
+          case (k, edge) =>
+            val srcNode = edge.srcNode
+            val (srcArea, srcMinTemp, srcMaxTemp) = (srcNode.area, srcNode.minTemp, srcNode.maxTemp)
+            val isSrcNodeACloud = (srcArea >= nodeMinArea) ||
+              (srcArea < nodeMinArea && (srcMinTemp / srcMaxTemp) < convectiveFraction)
+
+            val destNode = edge.destNode
+            val (destArea, destMinTemp, destMaxTemp) = (destNode.area, destNode.minTemp, destNode.maxTemp)
+            val isDestNodeACloud = (destArea >= nodeMinArea) ||
+              (destArea < nodeMinArea && (destMinTemp / destMaxTemp) < convectiveFraction)
+            var meetsOverlapCriteria = true
+            if (isSrcNodeACloud && isDestNodeACloud) {
+              val areaOverlap = edge.areaOverlap
+              val percentAreaOverlap = math.max((areaOverlap / srcArea), (areaOverlap / destArea))
```

Comment:
This line needs to be :
              val percentAreaOverlap = math.max((areaOverlap.toDouble / srcArea.toDouble), (areaOverlap.toDouble / destArea.toDouble))